File: Injections.NullModules/NullDHT11.cs
using System;
using TinyIoC;
using ACRemote;

namespace Injections.NullModules {
	public class NullDHT11 : Defaultable, IDHT11 {
		public static void register(TinyIoCContainer kern) {
			if (!non_default_exists<IDHT11>(typeof(NullDHT11))) {
				kern.Register<IDHT11, NullDHT11>().AsSingleton();
			}
		}
		public NullDHT11() {
			
		}
		public double temp {
			get {
				return 25;
			}
		}
		public double humidity {
			get {
				return 80;
			}
		}
	}
}

File: Injections.NullModules/NullDHT11.vb
' Null DHT11 module provides get/set of variables, no actual temp/humidity
' Author: Kevin Ross

Imports TinyIoC
Imports ACRemote

Public Class NullDHT11
    Inherits Defaultable
    Implements IDHT11
    Private temp_ As Double, humid_ As Double
    Public Sub New()
    End Sub
    Public Property temperature As Double Implements IDHT11.temperature
        Get
            Console.WriteLine(String.Format("NullDHT11: Get Temp: {0}", temp_))
            Return temp_
        End Get
        Set(value As Double)
            Console.WriteLine(String.Format("NullDHT11: Set Temp: {0}", value))
            temp_ = value
        End Set
    End Property
    Public Property humidity As Double Implements IDHT11.humidity
        Get
            Console.WriteLine(String.Format("NullDHT11: Get Humidity: {0}", humid_))
            Return humid_
        End Get
        Set(value As Double)
            Console.WriteLine(String.Format("NullDHT11: Set Humidity: {0}", value))
            humid_ = value
        End Set
    End Property
End Class